easybuild/framework/extensioneasyblock.py

```diff
@@ -106,10 +106,7 @@ def _set_start_dir(self):
         Uses existing value of self.start_dir if it is already set, an absolute path and exists
         otherwise use self.ext_dir (path to extracted source) as base dir if that is set and exists.
```

Comment:
```suggestion
        Use existing value of self.start_dir if it is already set and return the corresponding absolute path
        For relative paths use self.ext_dir (path to extracted source) as base dir
        Non-extracted extensions (e.g. Python WHL) have a None start_dir
```

Reply:
Good suggestion! However "Non-extracted extensions (e.g. Python WHL) have a None start_dir" does not hold according to this function: They have an unset `ext_dir` and hence `start_dir` is not changed. I hence used
```
        Uses existing value of self.start_dir if it is set.
        If self.ext_dir (path to extracted source) is set, it is used as the base dir for relative paths.
        Otherwise (e.g. for non-extracted extensions like Python WHL files) or when the computed start_dir
        does not exist the start dir is not changed.
```

Of course we could further refine this to use the CWD to convert relative paths to absolute ones (I added tests: `start_dir` might be set to `.` or `..` in the EC) but not sure if that is worth it. But then we would have the invariant that the start_dir is always absolute

Reply:
I agree that it would be nice to make `start_dir` always an absolute path, that was the intention of my initial PR. In the case where `self.start_dir=None` there is no `self.ext_dir` either, right? So one option would be to set `start_dir` to the absolute path of `builddir`. That will exist at this stage.

Reply:
I wouldn't set start_dir if both `self.start_dir` and `self.ext_dir` are None. Accessing it in that case is likely an error and shouldn't be masked by setting it (kind of) arbitrary.

However I would agree to resolving start_dir relative to the CWD when it is set and ext_dir is not. That keeps the current behavior (using builddir would potentially change current behavior) and we have the invariant that it is either None or absolute